test_data/std/bus_permutation_via_challenges_ext.asm

```diff
@@ -29,42 +20,10 @@ machine Main with degree: 8 {
 
     let permutation_constraint = first_four $ [a1, a2] is (1 - first_four) $ [b1, b2];
 
-    // TODO: Functions currently cannot add witness columns at later stages,
-    // so we have to manually create it here and pass it to permutation(). 
-    col witness stage(1) z1;
-    col witness stage(1) z2;
-    let z = Fp2::Fp2(z1, z2);
-
-    col witness stage(1) u1;
-    col witness stage(1) u2;
-    let u = Fp2::Fp2(u1, u2);
-
-    permutation(1, [z1,z2], [u1, u2], alpha, beta, permutation_constraint);
-
-    let is_first: col = std::well_known::is_first;
-    
-    col witness stage(1) z1_next;
-    col witness stage(1) z2_next;
-    query |i| {
-        let hint_send = compute_next_z_send_permutation(is_first, 1, z, alpha, beta, permutation_constraint);
-        std::prover::provide_value(z1_next, i, hint_send[0]);
-        std::prover::provide_value(z2_next, i, hint_send[1]);
-    };
-
-    z1' = z1_next;
-    z2' = z2_next;
-
-    col witness stage(1) u1_next;
-    col witness stage(1) u2_next;
-    query |i| {
-        let hint_receive = compute_next_z_receive_permutation(is_first, 1, u, alpha, beta, permutation_constraint);
-        std::prover::provide_value(u1_next, i, hint_receive[0]);
-        std::prover::provide_value(u2_next, i, hint_receive[1]);
-    };
-
-    u1' = u1_next;
-    u2' = u2_next;
+    permutation(1, permutation_constraint);
 
-    is_first' * (z1 + u1) = 0;
-    is_first' * (z2 + u2) = 0;
+    //let is_first: col = std::well_known::is_first;
+//
+    //is_first' * (z1 + u1) = 0;
+    //is_first' * (z2 + u2) = 0;
```

Comment:
@chriseth Any idea how I can simplify this part to use the accumulators, which are inside the bus_interaction, also for the permutation_via_bus so that I can use these constraints `is_first' * (z1 + u1) = 0;` and
    `is_first' * (z2 + u2) = 0;`

Reply:
Because z and u are the accumulators, which I’m going to move them into bus_interaction function. But the permutation function consists of just bus_send(); and bus_receive(); (so accumulators are used inside de bus_interaction, not in the permutation function)